### PR TITLE
DFG CRUD behavior standardization

### PR DESCRIPTION
**Pull request by GearsAD**

### CODE REVIEW

**Comment by dehann** [comment]
pending decisions from #134 disccussion

File: src/services/AbstractDFG.jl [hunk 1]
```diff
@@ -84,82 +87,92 @@ end
...
 	error("updateFactor! not implemented for $(typeof(dfg))")
 end
 
 """
     $(SIGNATURES)
 Delete a DFGVariable from the DFG using its label.
+Returns: True if deleted, false if not.
 """
-function deleteVariable!(dfg::G, label::Symbol)::AbstractDFGVariable where G <: AbstractDFG
+function deleteVariable!(dfg::G, label::Symbol)::Bool where G <: AbstractDFG
```
Comment (dehann) [inline src/services/AbstractDFG.jl]: did we come to a decision delete being bool or union in the end?

Comment (dehann) [reply src/services/AbstractDFG.jl]: maybe there is a pointer to the decision on this?  but my expectation would be give the just deleted value back one last time...

**Comment by GearsAD** [comment]
Yep, we need to work more on this for v0.6

**Comment by Affie** [comment]
Return type decision was change in #145 and #134 Implemented in 0.6 branch and merged to develop

**Pull request closed without merging**